src/components/tree/reactD3Interface/change.js

```diff
@@ -92,6 +92,18 @@ export const changePhyloTreeViaPropsComparison = (mainTree, phylotree, oldProps,
     args.svgHasChangedDimensions = true;
   }
 
+  /* split tree by colored-by trait */
+  /* just copies into tree; will be null/empty string/not defined if no trait */
+  if (newProps.tree.splitTreeByTrait !== oldProps.tree.splitTreeByTrait)
+  {
+    args.splitTreeByTrait = newProps.tree.splitTreeByTrait;
+    if (args.splitTreeByTrait == null) {// if the split is being unset, reset the layout
+    // todo: could use updateLayout instead?
```

Comment:
Let's revisit this, and exactly how we're using d3 to update the DOM (e.g. transitions) once the algorithm is working correctly

Reply:
Update on this - the d3 transition is working reasonably well for turning the toggle off. For turning it on, it still looks a bit jerky.

Reply:
@jameshadfield, if you feel we are at this point, do you have any thoughts on why the transition is still not smooth when enabling the toggle (though it is smooth at all other times)?

Reply:
I'll look into this shortly, as I do think we're at this point 😄